src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -20096,5 +20096,83 @@ public void MethodA(ClassC obj)
         }
+
+        [Fact]
+        public void PartialMethodsConsiderRefKindDifferences_NoneWithRef()
+        {
+            CreateCompilation(@"
+partial class C {
+    partial void M(int i);
+    partial void M(ref int i) {}  
+}").VerifyDiagnostics(
+                // (4,18): error CS0759: No defining declaration found for implementing declaration of partial method 'C.M(ref int)'
+                //     partial void M(ref int i) {}  
+                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "M").WithArguments("C.M(ref int)").WithLocation(4, 18));
+        }
+
+        [Fact]
+        public void PartialMethodsConsiderRefKindDifferences_NoneWithIn()
+        {
+            CreateCompilation(@"
+partial class C {
+    partial void M(int i);
+    partial void M(in int i) {}  
+}").VerifyDiagnostics(
+                // (4,18): error CS0759: No defining declaration found for implementing declaration of partial method 'C.M(in int)'
+                //     partial void M(in int i) {}  
+                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "M").WithArguments("C.M(in int)").WithLocation(4, 18));
+        }
+
+        [Fact]
+        public void PartialMethodsConsiderRefKindDifferences_RefWithNone()
+        {
+            CreateCompilation(@"
+partial class C {
+    partial void M(ref int i);
+    partial void M(int i) {}  
+}").VerifyDiagnostics(
+                // (4,18): error CS0759: No defining declaration found for implementing declaration of partial method 'C.M(int)'
+                //     partial void M(int i) {}  
+                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "M").WithArguments("C.M(int)").WithLocation(4, 18));
+        }
+
+        [Fact]
+        public void PartialMethodsConsiderRefKindDifferences_RefWithIn()
+        {
+            CreateCompilation(@"
+partial class C {
+    partial void M(ref int i);
+    partial void M(in int i) {}  
+}").VerifyDiagnostics(
+                // (4,18): error CS0759: No defining declaration found for implementing declaration of partial method 'C.M(in int)'
+                //     partial void M(in int i) {}  
+                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "M").WithArguments("C.M(in int)").WithLocation(4, 18));
+        }
+
+        [Fact]
+        public void PartialMethodsConsiderRefKindDifferences_InWithNone()
+        {
+            CreateCompilation(@"
+partial class C {
+    partial void M(in int i);
+    partial void M(int i) {}  
+}").VerifyDiagnostics(
+                // (4,18): error CS0759: No defining declaration found for implementing declaration of partial method 'C.M(int)'
+                //     partial void M(int i) {}  
+                Diagnostic(ErrorCode.ERR_PartialMethodMustHaveLatent, "M").WithArguments("C.M(int)").WithLocation(4, 18));
+        }
+
+        [Fact]
+        public void PartialMethodsConsiderRefKindDifferences_InWithRef()
```

Comment:
>PartialMethodsConsiderRefKindDifferences_InWithRef [](start = 20, length = 50)

Should probably fold ```out``` into the test matrix.

Reply:
Partial methods cannot have `out` parameters, and there exists already a test for that.
What is the scenario you're proposing?

---
In reply to: [178658268](https://github.com/dotnet/roslyn/pull/25893#discussion_r178658268) [](ancestors = 178658268)

Reply:
Discussed offline. Either confirming CS759 is reported, or lookup the symbol table and confirm they're not merged.

---
In reply to: [178992135](https://github.com/dotnet/roslyn/pull/25893#discussion_r178992135) [](ancestors = 178992135,178658268)